### ISSUE TITLE
Use variable declared as `const` as literal type

### ISSUE THREAD

**Issue by Lenne231** [Microsoft/TypeScript#12603]
It would be nice to be able to use a variable declared as `const` as a literal type in a type annotation.

**TypeScript Version:**  2.0.0

**Code**

```ts
export const INCREMENT = 'INCREMENT';
export const DECREMENT = 'DECREMENT'
export const ADD = 'ADD';

type Action =
      { type: INCREMENT }
    | { type: DECREMENT };
```

**Expected behavior:**
No error, type `Action` is `{ type: 'INCREMENT' } | { type: 'DECREMENT' }`.

**Actual behavior:**
Error: Cannot find name `INCREMENT`

**Comment by Lenne231** [comment]
Ok,

```typescript
export const INCREMENT : 'INCREMENT' = 'INCREMENT';
export const DECREMENT : 'DECREMENT' = 'DECREMENT';

type Action =
      { type: typeof INCREMENT }
    | { type: typeof DECREMENT };
```

seems to work, but why is `const INCREMENT = 'INCREMENT';` of type `string` and not of type `INCREMENT`?

**Comment by ahejlsberg** [comment]
This is fixed in 2.1 by #10676.
